src/Location/Location.tsx

```diff
@@ -12,107 +12,115 @@ import wind from '../assets/wind.png'
 // feel free to break up lines of text for readability, i'm getting tired
 
 import { getAllData } from '../helpers/dataCleaner'
+import { LocationProps } from '../helpers/locationProps'
 
-interface LocationProps {
-		date?: string,
-		day?: string,
-		time?: number,
-		city?: string;
-		state?: string,
-		country?: string,
-		temp?: number,
-		aqi?: number,
-		uvIndex?: number,
-		visibility?: number,
-		wind?: number,
-		windDirection?: string,
-		precipitation?: number,
-		allergyLevel?: number,
-		discussion?: string,
-  }
 
-// const Location = ({locationData, currentWeather, weeklyWeather, currentAir}) => {
-// 	const [allData, setAllData] = useState()
 
-// 	// setAllData(getAllData('denver'))
-	
-// 	return (
-// 		<section className="location-section">
-// 			<h2 className='current-city'>{locationData.locality}</h2>
-// 			<h3 className='current-region'>{locationData.region}, {locationData.country_code}</h3>
-// 			<div className='info-box'>
-// 				<h4 className='info-box-header'>HAPPENING NOW</h4>
-// 				<p className='current-date'>{Date.now().toLocaleString()}</p>
-// 				<article className='current-weather'>
-// 					<div className='current-weather-left'>
-// 						<h5 className='current-temp'>{currentWeather.temp}&deg;</h5>
-// 						<img src={cloudyNight} alt='clouds covering moon for current sky' className='large-weather-icon'/>
-// 					</div>
-// 					<div className='current-weather-right'>
-// 						<span className='current-weather-right-top'>
-// 							<img src={wind} alt='wind icon for wind speed direction' className='small-weather-icon'/> 
-// 							<p className='current-wind'>{currentWeather.wind_speed} mph / {currentWeather.wind_direction}</p>
-// 						</span>
-// 						<span className='current-weather-right-bottom'>
-// 							<img src={waterdrop} alt='rain droplet icon for precipitation' className='small-weather-icon'/> 
-// 							<p className='current-precipitation'>{(currentWeather.precipitation) ? currentWeather.precipitation : 0}</p>
-// 						</span>
-// 					</div>
+const Location = ({locationData, currentWeather, weeklyWeather, currentAir}: LocationProps) => {
```

Comment:
Thanks for accounting for all the props here. What do you think about making the fetch call for the data in Location instead of App? Because we don't have to pass the data we get back into any other component, the call could live here. It would mean that the only props passed into Location would be the coordinates of the selected result and the query. Lets hop on a call soon and I can show you what I mean!

Reply:
I was actually thinking the same thing it would be a quick adjustment.